### PR TITLE
Tweak the Info struct

### PR DESCRIPTION
**Pull request by HeroicKatora**
It is now non-exhaustive which simplifies adding more information in the future. The borrowed option is also available for the icc chunk. The `InfoFilterType`, which has long been replaced for its encoding purpose, is removed.

### CODE REVIEW

File: src/encoder.rs [hunk 1]
```diff
@@ -415,16 +410,15 @@ impl PartialInfo {
...
-        Info {
-            width: self.width,
-            height: self.height,
-            bit_depth: self.bit_depth,
-            color_type: self.color_type,
-            frame_control: self.frame_control,
-            animation_control: self.animation_control,
-            compression: self.compression,
-            ..Default::default()
-        }
```
Comment (fintelia) [inline src/encoder.rs:427]: Was this change actually necessary, or is it just for stylistic reasons?

Comment (HeroicKatora) [reply src/encoder.rs:427]: Necessary, as the `..Default::default()` syntax is incompatible with private members.

**Review by fintelia — APPROVED** [review]
